### PR TITLE
feat: capture gracefully exceptions

### PR DESCRIPTION
**Pull request by spirosoik**

### CODE REVIEW

File: tracer.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+	tracerProvider := sdktrace.NewTracerProvider(
+		sdktrace.WithSyncer(exporter),
+		sdktrace.WithResource(newResource(ctx,
+			attribute.String("event", string(retTracer.eventData)),
+		)),
+	)
+	retTracer.provider = tracerProvider
+	otel.SetTracerProvider(tracerProvider)
+	panic("skata")
```
Comment (nirLumigo) [inline tracer.go:52]: What does it mean? Why do we panic here?

File: wrapper.go [hunk 2]
```diff
@@ -51,62 +48,24 @@ func WrapHandler(handler interface{}, conf *Config) interface{} {
...
+		tracer.Start()
 
 		response, lambdaErr := otellambda.WrapHandler(lambda.NewHandler(handler),
-			otellambda.WithTracerProvider(tracerProvider),
-			otellambda.WithFlusher(tracerProvider)).Invoke(traceCtx, payload)
+			otellambda.WithTracerProvider(tracer.provider),
+			otellambda.WithFlusher(tracer.provider)).Invoke(tracer.traceCtx, payload)
 
-		os.Setenv("IS_WARM_START", "true") // nolint
+		tracer.End(response, lambdaErr)
```
Comment (nirLumigo) [inline wrapper.go:67]: 💪

**Review by nirLumigo — CHANGES_REQUESTED** [review]
Nice! :)

File: _example/lambda.go [hunk 3]
```diff
@@ -83,10 +33,9 @@ func HandleRequest(ctx context.Context, name MyEvent) (string, error) {
 
 func main() {
 	os.Setenv("LUMIGO_DEBUG", "true")
-	wrappedHandler := lumigotracer.WrapHandlerWithAWSConfig(HandleRequest, &lumigotracer.Config{
+	wrappedHandler := lumigotracer.WrapHandler(HandleRequest, &lumigotracer.Config{
 		PrintStdout: false,
```
[the remark this answers is not in the document]
Comment (spirosoik) [reply _example/lambda.go:37]: yes of course we can. the print stdout is to help debug the span mostly

File: tracer.go (hunk 1 above)
Comment (spirosoik) [reply tracer.go:52]: hahaha actually that was testing. removing

**Review by nirLumigo — APPROVED** [review]

**Pull request merged**